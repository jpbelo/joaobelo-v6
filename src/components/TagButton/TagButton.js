import React, { Component } from 'react'
import PropTypes from 'prop-types'
import styled from 'styled-components'

const TagButtonContainer = styled.a`
  font-size: 11px;
  display: inline-block;
  border-radius: 6px;
  padding: 0 10px;
  margin-right: 6px;
  margin-bottom: 6px;
  line-height: 2;
  background-color: ${(props) => props.theme.colors.grey.light};
  transition: background-color 250ms ease-in-out;
  &[href^='http'] {
    background-color: ${(props) => props.theme.colors.secondary.bgAlt};
    &:hover {
      background: ${(props) => props.theme.colors.primary.bgAlt};
    }
  }
`

export default class TagButton extends Component {
  render() {
    const { url, name } = this.props
    return (
      <TagButtonContainer target="_blank" href={url}>
        {name}
      </TagButtonContainer>
    )
  }
}

TagButton.propTypes = {
  name: PropTypes.string.isRequired,
  url: PropTypes.string.isRequired,
}
