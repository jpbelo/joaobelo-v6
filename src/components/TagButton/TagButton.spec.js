import React from 'react'
import 'jest-styled-components'

import { renderWithTheme } from 'helpers/renderWithTheme'
import TagButton from './TagButton'

const mockProps = {
  url: 'http://google.com',
  name: 'Google',
}

test('Button changes color when hovered', () => {
  let tree = renderWithTheme(<TagButton {...mockProps} />).toJSON()
  expect(tree).toMatchSnapshot()
})
