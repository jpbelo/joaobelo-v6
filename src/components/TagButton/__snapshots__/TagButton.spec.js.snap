// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button changes color when hovered 1`] = `
.c0 {
  font-size: 11px;
  display: inline-block;
  border-radius: 6px;
  padding: 0 10px;
  margin-right: 6px;
  margin-bottom: 6px;
  line-height: 2;
  background-color: hsl(0,0%,90%);
  -webkit-transition: background-color 250ms ease-in-out;
  transition: background-color 250ms ease-in-out;
}

.c0[href^='http'] {
  background-color: hsl(46,50%,92%);
}

.c0[href^='http']:hover {
  background: hsl(0,50%,90%);
}

<a
  className="c0"
  href="http://google.com"
  target="_blank"
>
  Google
</a>
`;
